clipper_admin/clipper_admin/clipper_admin.py

```diff
@@ -0,0 +1,995 @@
+            The name of the deployed model
+        version : str
+            The version to assign this model. Versions must be unique on a per-model
+            basis, but may be re-used across different models.
+        input_type : str
+            The type of the request data this endpoint can process. Input type can be
+            one of "integers", "floats", "doubles", "bytes", or "strings". See the
+            `User Guide <http://clipper.ai/user_guide/#input-types>`_ for more details
+            on picking the right input type for your application.
+        image : str, optional
+            A docker image name. If provided, the image will be recorded as part of the
+            model descrtipin in Clipper when registering the model but this method will
+            make no attempt to launch any containers with this image.
+        labels : list(str), optional
+            A list of strings annotating the model. These are ignored by Clipper
+            and used purely for user annotations.
+
+        Raises
+        ------
+        :py:exc:`clipper.UnconnectedException`
+        :py:exc:`clipper.ClipperException`
+        """
+
+        if not self.connected:
+            raise UnconnectedException()
+        version = str(version)
+        url = "http://{host}/admin/add_model".format(
+            host=self.cm.get_admin_addr())
+        if image is None:
+            image = CONTAINERLESS_MODEL_IMAGE
+        if labels is None:
+            labels = DEFAULT_LABEL
+        req_json = json.dumps({
+            "model_name": name,
+            "model_version": version,
+            "labels": labels,
+            "input_type": input_type,
+            "container_name": image,
+            "model_data_path": "DEPRECATED",
+        })
+        headers = {'Content-type': 'application/json'}
+        logger.debug(req_json)
+        r = requests.post(url, headers=headers, data=req_json)
+        if r.status_code != requests.codes.ok:
+            msg = "Received error status code: {code} and message: {msg}".format(
+                code=r.status_code, msg=r.text)
+            logger.error(msg)
+            raise ClipperException(msg)
+
+    def get_num_replicas(self, name, version):
+        """Gets the current number of model container replicas for a model.
+
+        Parameters
+        ----------
+        name : str
+            The name of the model
+        version : str, optional
+            The version of the model. If no version is provided,
+            the currently deployed version will be used.
+
+        Returns
+        -------
+        int
+            The number of active replicas
+
+        Raises
+        ------
+        :py:exc:`clipper.UnconnectedException`
+        """
+        if not self.connected:
+            raise UnconnectedException()
+        if version is None:
+            model_info = self.get_all_models(verbose=True)
+            for m in model_info:
+                if m["is_current_version"]:
```

Comment:
I think this needs to be filtered by the supplied model name. Otherwise, we're setting `version` to be the current version of the last model contained in the dictionary.

Reply:
Oh yeah. Good catch.